RIOT-OS/RIOT
# cpu/sam0: add support for SAMD5x/SAME5x

### Contribution description
This adds support for the Atmel SAMD51 & SAME54 Cortex M4F MCU from the SAM D5x/E5x family and the SAM E54 Xplained Pro Evaluation Board.

The SAMD5x family shares it's peripherals with other sam0 MCUs like the saml1x on which most of this code is based.

So far, UART, I2C, SPI, Timer, RTC and GPIOs have been tested.

There is an awful lot of copy and paste going on between `saml1x/periph` and `same54/periph` with only minor differences in `pm.c` and `timer.c` (SAMD5x supports more sleep states and needs to combine two timers to make a 32bit timer).

### Testing procedure
Flash an example application (e.g. `examples/default` or `examples/timer_periodic_wakeup`) using `make BOARD=same54-xpro flash term` and observe that the application behaves as expected.
